addons/mass_mailing/models/utm_campaign.py

```diff
@@ -152,7 +144,7 @@ def _cron_process_mass_mailing_ab_testing(self):
         ab_testing_campaign = self.search([
             ('ab_testing_schedule_datetime', '<=', fields.Datetime.now()),
             ('ab_testing_winner_selection', '!=', 'manual'),
-            ('ab_testing_completed', '=', False),
+            ('ab_testing_winner_mailing', '=', False),
```

Comment:
Doesn't this require a migration script?

As the "ab_testing_winner_mailing" will be added as False for existing databases, it means that the CRON will re-evaluate all of them, right?

We should probably find a way to assign the "ab_testing_winner_mailing" for the existing campaigns then.

Reply:
Ah yeah I hadn't thought of that.. Do we even want to keep this field at that point? I guess the mailings could just have a related on ``ab_testing_winner_mailing``, we're going to fetch it from the db either way so it shouldn't be more expensive.

Reply:
I think we need to keep an easy way of knowing that a campaign is "done" and being able to search for it.
This is important for the CRON and also used at a few places in the code.

One other way of doing it is to make "ab_testing_completed" computed stored, which lowers the impact as we should be able to keep this code exactly the same as before.
It could be fine as I don't expect that we will have a ton of mass mailing records.

Maybe I did not understand your question correctly?

Reply:
Basically I was just thinking this:

All `ab_testing_completed` does now is check whether `ab_testing_winner_mailing` is Truthy or Falsy.  So why not remove it entirely, and just check the boolean value of `ab_testing_winner_mailing` everywhere it's used instead.

This would avoid an otherwise pointless compute. But if you feel it's not straight-forward enough I can leave it in.

Reply:
It's a bit better in terms of readability to be able to see "ab_testing_completed", but both work, I'm fine with either having it computed or removed. It's readability vs a bit of overhead, I let you pick what you prefer :slightly_smiling_face: 

The big take on this comment is that we need a migration script, no matter what we pick, as we need to set "ab_testing_winner_mailing" for existing records :smile:

Reply:
Ok, I prefer readability of course :D

I was under the impression we wanted to keep the number of fields as low as possible even for non-stored fields, but I guess it's not really an issue for such simple computations.

Added the field on my upgrade branch. And created a PR for it while I was at it..

Reply:
Hello @thiryr 

Just checked the upgrade PR and it seems very short.

What I meant is that we will have to **fill in** the new ab_testing_winner_mailing for all completed campaigns to avoid that the CRON considers them all as "not completed".
Just removing 2 fields will not be enough.

Maybe you did not push the latest changes? Or I clicked a wrong link? Or I missed something?

Side-note: Also it should be named "ab_testing_winner_mailing_id" with "_id" at the end as it's a m2o field.